frontend/catalyst/compilation_pipelines.py

```diff
@@ -524,38 +517,27 @@ def compile(self):
         if self.compiled_function and self.compiled_function.shared_object:
             self.compiled_function.shared_object.close()
 
-        if self.compiling_from_textual_ir:
-            # Module name can be anything.
-            module_name = "catalyst_module"
-            shared_object, llvm_ir, inferred_func_data = self.compiler.run_from_ir(
-                self.user_function, module_name, self.workspace
-            )
-            qfunc_name = inferred_func_data[0]
-            # Parse back the return types given as a semicolon-separated string
-            with ir.Context():
-                restype = [ir.RankedTensorType.parse(rt) for rt in inferred_func_data[1].split(",")]
-        else:
-            # This will make a check before sending it to the compiler that the return type
-            # is actually available in most systems. f16 needs a special symbol and linking
-            # will fail if it is not available.
-            #
-            # WARNING: assumption is that the first function
-            # is the entry point to the compiled program.
-            entry_point_func = self.mlir_module.body.operations[0]
-            restype = entry_point_func.type.results
-
-            for res in restype:
-                baseType = ir.RankedTensorType(res).element_type
-                mlir_type_to_numpy_type(baseType)
-
-            # The function name out of MLIR has quotes around it, which we need to remove.
-            # The MLIR function name is actually a derived type from string which has no
-            # `replace` method, so we need to get a regular Python string out of it.
-            qfunc_name = str(self.mlir_module.body.operations[0].name).replace('"', "")
-
-            shared_object, llvm_ir, inferred_func_data = self.compiler.run(
-                self.mlir_module, self.workspace
-            )
+        # This will make a check before sending it to the compiler that the return type
+        # is actually available in most systems. f16 needs a special symbol and linking
+        # will fail if it is not available.
+        #
+        # WARNING: assumption is that the first function
+        # is the entry point to the compiled program.
+        entry_point_func = self.mlir_module.body.operations[0]
+        restype = entry_point_func.type.results
+
+        for res in restype:
+            baseType = ir.RankedTensorType(res).element_type
+            mlir_type_to_numpy_type(baseType)
```

Comment:
```suggestion
        # WARNING: assumption is that the first function
        # is the entry point to the compiled program.
        entry_point_func = self.mlir_module.body.operations[0]
        restype = entry_point_func.type.results

        for res in restype:
            baseType = ir.RankedTensorType(res).element_type
            # This will make a check before sending it to the compiler that the return type
            # is actually available in most systems. f16 needs a special symbol and linking
            # will fail if it is not available.
            mlir_type_to_numpy_type(baseType)
```